### DISCUSSION TITLE
Ways to help the project

### DISCUSSION THREAD

**Discussion by cstrlcs** [WayfireWM/wayfire#1223] [General]
Hello!

First of all, I really like Wayfire and I appreciate all the hard work involved.

I saw that the youtube channel has not been updated for a while, I'm afraid is there any chance of this project becoming unmantained in the near future?

I'd really like to contribute to it but, since I'm really bad at c++, I want to know if there are plans to create a patreon or a donation channel, so I could at least pay a coffee for the devs. (:

Thanks!

**Comment by ammen99** [comment]
Hi, thanks for reaching out!

Fortunately, the youtube channel is a bit misleading. I have not uploaded any new videos, but this is because most of the new changes are bug fixes, code improvements and other features which are hard to visualize. But, development is still ongoing, and I don't think the project will become unmaintained in the near future, don't worry :)

On topic of donations, no, there is currently no way to donate.


 If you want to contribute, C++ is not the only way (I know this is becoming cliche on the internet, but it is true). Our wiki can be improved/expanded, some options need to be updated - see #1070 for example. Managing bugs is also something people without much coding knowledge could do - verifying whether new crash reports have a good stack trace, clear reproduction steps, etc. Even simply regularly testing git version of Wayfire and reporting regressions is extremely valuable, too.

Finally, if you do want to learn C++: contributing is a good way to do that! For most bugs, the time-consuming part is understanding what exactly is going wrong. To figure it out, you need to understand a lot of the code, maybe add a few debugging logs, and finally write a (hopefully) short bug fix, which we could help you with :) If you, or anyone else wants to do this, do not hesitate to ask questions about the codebase or C++, I'll try my best to help, and on Matrix/IRC there are several other people which can help, too.

**Reply by cstrlcs** [reply]
Hi, thank you for your answer!

I'll definitely look into the code and start tracking bugs. I'll try to improve my C++ on this so I can contribute even more.
Also, I'll take a look into the Wiki and see where I can help.

Thank you for your nice work, again.
